src/jbms/argparse.cpp

```diff
@@ -19,15 +19,116 @@ namespace argparse {
 
 using std::weak_ptr;
 
-string repr(string_view s) {
-  // FIXME: make this escape bad characters
-  string x;
-  x += '\'';
-  x += string(s);
-  x += '\'';
-  return x;
+namespace {
+void chr_cstrliteral(unsigned char ch, char *buffer, std::size_t buflen) {
+  /**
+   * @brief Escapes a single character @p ch and appends its escaped value onto
+   * the buffer. This function must take an @a __uint8_t type otherwise we will
+   * run into buffer overflow. Also, @p buffer must be a @a char* to the end of
+   * the buffer escaped so far inorder to write to the buffer using C-library
+   * functions.
+   * @param ch Character to escape (if a control character).
+   * @param buffer Buffer to store the escaped value in.
+   * @param buflen Buffer length.
+   */
+  if (buflen < 2) {
+    *buffer = '\0';  // only null-terminator can be stored
+  } else if ((ch >= 32 && ch <= 126) && ch != '\'' && ch != '\"' &&
+             ch != '\\') {
+    sprintf(
+        buffer, "%c",
+        ch);  // opposite of control characters can be saved just as they are
+  } else if (buflen < 3) {
+    *buffer = '\0';  // just add a null-terminator; don't need to go further
+  } else {
+    switch (ch) {
+      case '\a':
+        strcpy(buffer, "\\a");
+        break;
+      case '\b':
+        strcpy(buffer, "\\b");
+        break;
+      case '\f':
+        strcpy(buffer, "\\f");
+        break;
+      case '\n':
+        strcpy(buffer, "\\n");
+        break;
+      case '\r':
+        strcpy(buffer, "\\r");
+        break;
+      case '\t':
+        strcpy(buffer, "\\t");
+        break;
+      case '\v':
+        strcpy(buffer, "\\v");
+        break;
+      case '\\':
+        strcpy(buffer, "\\\\");
+        break;
+      case '\'':
+        strcpy(buffer, "\\'");
+        break;
+      case '\"':
+        strcpy(buffer, "\\\"");
+        break;
+      default:
+        if (buflen < 5)
+          *buffer = '\0';
+        else
+          sprintf(buffer, "\\%03o",
+                  ch);  // this handles control characters that are not
+                        // printable and does not meet any case
+        break;
+    }
+  }
 }
 
+void str_cstrliteral(const std::string &str, std::string &buffer,
+                     std::size_t reserved_bits) {
+  /**
+   * @brief Escapes the special string @p str and stores inside of the @p
+   * buffer.
+   * @param str Special characters string.
+   * @param buffer Buffer to store the escaped value in.
+   * @param buflen Length of the buffer.
+   */
+  // FIXME: Make this escape NULL bytes as well
+  for (const auto &ch : str) {
+    char *buf =
+        new char[buffer.length() +
+                 1];  // need to create a buffer inorder to write to the
+                      // memory inside the above function i.e., chr_cstrliteral()
```

Comment:
A plain `buffer` should be created inside the function `str_cstrliteral()` and a pointer to the end of the `buffer` should be transferred to the function `chr_cstrliteral()` in-order to write the escaped value of the selected character in this iteration.